pulsar-broker/src/main/java/org/apache/pulsar/broker/ManagedLedgerClientFactory.java

```diff
@@ -59,7 +59,11 @@ public void initialize(ServiceConfiguration conf, MetadataStoreExtended metadata
         managedLedgerFactoryConfig.setMaxCacheSize(conf.getManagedLedgerCacheSizeMB() * 1024L * 1024L);
         managedLedgerFactoryConfig.setCacheEvictionWatermark(conf.getManagedLedgerCacheEvictionWatermark());
         managedLedgerFactoryConfig.setNumManagedLedgerSchedulerThreads(conf.getManagedLedgerNumSchedulerThreads());
-        managedLedgerFactoryConfig.setCacheEvictionFrequency(conf.getManagedLedgerCacheEvictionFrequency());
+        // compatible logic : if managedLedgerCacheEvictionFrequency is configured,
+        long managedLedgerCacheEvictionIntervalMs = conf.getManagedLedgerCacheEvictionIntervalMs() > 0
```

Comment:
We can override `getManagedLedgerCacheEvictionIntervalMs` method to handle this compatibility issue like most other config fields does.

Reply:
you're right